packages/admin/admin/src/FinalForm.tsx

```diff
@@ -177,11 +179,16 @@ export function FinalForm<FormValues = AnyObject, InitialFormValues = Partial<Fo
 
                     if (!validationWarnings) {
                         registeredFields.forEach((fieldName) => {
-                            setFieldData(fieldName, { warning: undefined });
+                            if (formWarningsRef.current[fieldName]) {
+                                setFieldData(fieldName, { warning: undefined });
+                            }
```

Comment:
This was hard to understand. We should either create a dedicated variable `hasFormLevelWarning` or add a comment explaining that this prevents resetting field-level warnings.